# Background

E-commerce is a service that helps make it easier for people to buy their needs. Various products ranging from beauty products to electronic gadgets are sold in e-commerce services at the palm of your hand. Customers are also facilitated with various payment options ranging from COD (Cash on Delivery) to digital payments such as internet banking and e-wallet. Payment methods are one of the success factors in business transactions. Therefore, the company will analyze the efficiency of the payment methods that have been used in the transaction process

# Problem Question

How does the payment method perform based on transaction success and nominal income?

#Data Preparation

## Data Understanding

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
df = pd.read_csv('Pakistan Largest Ecommerce Dataset.csv')
df

<ipython-input-2-ec8985d3392f>:1: DtypeWarning: Columns (1,2,3,7,8,9,11,12,13,14,17,18,19) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('Pakistan Largest Ecommerce Dataset.csv')


item_id          status created_at  \
0        211131.0        complete   7/1/2016   
1        211133.0        canceled   7/1/2016   
2        211134.0        canceled   7/1/2016   
3        211135.0        complete   7/1/2016   
4        211136.0  order_refunded   7/1/2016   
...           ...             ...        ...   
1048570       NaN             NaN        NaN   
1048571       NaN             NaN        NaN   
1048572       NaN             NaN        NaN   
1048573       NaN             NaN        NaN   
1048574       NaN             NaN        NaN   

                                                       sku   price  \
0                                        kreations_YI 06-L  1950.0   
1        kcc_Buy 2 Frey Air Freshener & Get 1 Kasual Bo...   240.0   
2                                       Ego_UP0017-999-MR0  2450.0   
3                                           kcc_krone deal   360.0   
4                                              BK7010400AG   555.0   
...                                                    ...     ...   
1048570                                                NaN     NaN   
1048571                                                NaN     NaN   
1048572                                                NaN     NaN   
1048573                                                NaN     NaN   
1048574                                                NaN     NaN   

         qty_ordered  grand_total increment_id    category_name_1  \
0                1.0       1950.0    100147443    Women's Fashion   
1                1.0        240.0    100147444  Beauty & Grooming   
2                1.0       2450.0    100147445    Women's Fashion   
3                1.0         60.0    100147446  Beauty & Grooming   
4                2.0       1110.0    100147447            Soghaat   
...              ...          ...          ...                ...   
1048570          NaN          NaN          NaN                NaN   
1048571          NaN          NaN          NaN                NaN   
1048572          NaN          NaN          NaN                NaN   
1048573          NaN          NaN          NaN                NaN   
1048574          NaN          NaN          NaN                NaN   

        sales_commission_code  ...  Month Customer Since     M-Y    FY  \
0                          \N  ...    7.0         2016-7  7-2016  FY17   
1                          \N  ...    7.0         2016-7  7-2016  FY17   
2                          \N  ...    7.0         2016-7  7-2016  FY17   
3                 R-FSD-52352  ...    7.0         2016-7  7-2016  FY17   
4                          \N  ...    7.0         2016-7  7-2016  FY17   
...                       ...  ...    ...            ...     ...   ...   
1048570                   NaN  ...    NaN            NaN     NaN   NaN   
1048571                   NaN  ...    NaN            NaN     NaN   NaN   
1048572                   NaN  ...    NaN            NaN     NaN   NaN   
1048573                   NaN  ...    NaN            NaN     NaN   NaN   
1048574                   NaN  ...    NaN            NaN     NaN   NaN   

        Customer ID  Unnamed: 21  Unnamed: 22 Unnamed: 23 Unnamed: 24  \
0               1.0          NaN          NaN         NaN         NaN   
1               2.0          NaN          NaN         NaN         NaN   
2               3.0          NaN          NaN         NaN         NaN   
3               4.0          NaN          NaN         NaN         NaN   
4               5.0          NaN          NaN         NaN         NaN   
...             ...          ...          ...         ...         ...   
1048570         NaN          NaN          NaN         NaN         NaN   
1048571         NaN          NaN          NaN         NaN         NaN   
1048572         NaN          NaN          NaN         NaN         NaN   
1048573         NaN          NaN          NaN         NaN         NaN   
1048574         NaN          NaN          NaN         NaN         NaN  

In [ ]:
print(f'Jumlah awal baris pada dataset ini sebanyak {df.shape[0]} baris')
print(f'Jumlah awal kolom pada dataset ini sebanyak {df.shape[1]} kolom')

Jumlah awal baris pada dataset ini sebanyak 1048575 baris
Jumlah awal kolom pada dataset ini sebanyak 26 kolom


This dataset has data related to product info, payment methods, and status.
There are 21 named columns and 5 unnamed columns in this data, namely:

1. item_id = Transaction ID
2. status = Transaction Status
3. created_at = Transaction Date
4. sku = Product's Unique Code
5. price = Price per item
6. qty_ordered = Quantity Ordered
7. grand_total = Total Price
8. increment_id = Increment ID
9. category_name_1 = Prudoct's Category
10. sales_commision_code = Sales Commission Code
11. discount_amount = Discount Amount
12. payment_method = Payment Method
13. Working Date = Order Working Date
14. BI status = BI Status
15. MV = Total Order Price
16. Year = Order Year
17. Month = Order Month
18. Customer Since = Date of Customer Since
19. M-Y = Month - Year
20. FY = Fiscal Year
21. Customer ID = Customer ID


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1048575 entries, 0 to 1048574
Data columns (total 26 columns):
 #   Column                 Non-Null Count   Dtype  
---  ------                 --------------   -----  
 0   item_id                584524 non-null  float64
 1   status                 584509 non-null  object 
 2   created_at             584524 non-null  object 
 3   sku                    584504 non-null  object 
 4   price                  584524 non-null  float64
 5   qty_ordered            584524 non-null  float64
 6   grand_total            584524 non-null  float64
 7   increment_id           584524 non-null  object 
 8   category_name_1        584360 non-null  object 
 9   sales_commission_code  447349 non-null  object 
 10  discount_amount        584524 non-null  float64
 11  payment_method         584524 non-null  object 
 12  Working Date           584524 non-null  object 
 13  BI Status              584524 non-null  object 
 14   MV                    584524 non-

From the information above, we can find out the general value and data type of each column

In [ ]:
#unique value each column
pd.set_option('display.max_colwidth', -1)
listItem = []
for col in df.columns :
    listItem.append( [col, df[col].nunique(), df[col].unique()])

tableunique = pd.DataFrame(columns=['Column Name', 'Number of Unique', 'Unique Sample'],
                     data=listItem)
tableunique

<ipython-input-5-df1f1b986c81>:2: FutureWarning: Passing a negative integer is deprecated in version 1.0 and will not be supported in future version. Instead, use None to not limit the column width.
  pd.set_option('display.max_colwidth', -1)


Column Name  Number of Unique  \
0   item_id                584524             
1   status                 16                 
2   created_at             789                
3   sku                    84889              
4   price                  9121               
5   qty_ordered            74                 
6   grand_total            36829              
7   increment_id           408785             
8   category_name_1        16                 
9   sales_commission_code  7226               
10  discount_amount        28058              
11  payment_method         18                 
12  Working Date           789                
13  BI Status              4                  
14   MV                    9720               
15  Year                   3                  
16  Month                  12                 
17  Customer Since         26                 
18  M-Y                    26                 
19  FY                     3                  
20  Customer ID            115326             
21  Unnamed: 21            0                  
22  Unnamed: 22            0                  
23  Unnamed: 23            0                  
24  Unnamed: 24            0                  
25  Unnamed: 25            0                  

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Unique Sample  
0   [211131.0, 211133.0, 211134.0, 211135.0, 211136.0, 211137.0, 211138.0, 211139.0, 211140.0, 211141.0, 211142.0, 211143.0, 211144.0, 211145.0, 211146.0, 211147.0, 211149.0, 211150.0, 211151.0, 211152.0, 211153.0, 211154.0, 211155.0, 211156.0, 211157.0, 211158.0, 211160.0, 211162.0, 211163.0, 211164.0, 211166.0, 211168.0, 211169.0, 211170.0, 211171.0, 211172.0, 211173.0, 211174.0, 211175.0, 211176.0, 211177.0, 211179.0, 211180.0, 211182.0, 211184.0, 211185.0, 21

From the information above, we can find out the *Unique Value* of each column in this data

In [ ]:
#Percentage of missing value in each column
df.isnull().sum()/df.shape[0]*100

item_id                  44.255394 
status                   44.256825 
created_at               44.255394 
sku                      44.257302 
price                    44.255394 
qty_ordered              44.255394 
grand_total              44.255394 
increment_id             44.255394 
category_name_1          44.271034 
sales_commission_code    57.337434 
discount_amount          44.255394 
payment_method           44.255394 
Working Date             44.255394 
BI Status                44.255394 
 MV                      44.255394 
Year                     44.255394 
Month                    44.255394 
Customer Since           44.256443 
M-Y                      44.255394 
FY                       44.255394 
Customer ID              44.256443 
Unnamed: 21              100.000000
Unnamed: 22              100.000000
Unnamed: 23              100.000000
Unnamed: 24              100.000000
Unnamed: 25              100.000000
dtype: float64

From initial observations of the data above, we can conclude:

1. This dataframe has 26 columns and 1048575 rows
2. From all the data, there are missing values in each column with percentages ranging from 44.25% - 100%
3. There is a value \N in the status column, sales_commission_code, and category_name_1 which we can consider as nan or missing value
4. The Year, Month, and M-Y columns have the same value

## Data Cleansing

In this process, we will clean the dataset that we have. The first step is to change the column names to create consistency and make it easier to read the data.

In [ ]:
df = df.rename(columns={
    'item_id': 'ItemID',
    'status': 'Status',
    'created_at': 'CreatedAt',
    'sku': 'SKU',
    'price': 'Price',
    'qty_ordered': 'QuantityOrdered',
    'grand_total': 'GrandTotal',
    'increment_id': 'IncrementID',
    'category_name_1': 'CategoryName',
    'sales_commission_code': 'SalesCommissionCode',
    'discount_amount': 'DiscountAmount',
    'payment_method': 'PaymentMethod',
    'Working Date': 'WorkingDate',
    'BI Status': 'BIStatus',
    'MV': 'MV',
    'Year': 'Year',
    'Month': 'Month',
    'Customer Since': 'CustomerSince',
    'M-Y': 'MY',
    'FY': 'FY',
    'Customer ID': 'CustomerID'
})
df

ItemID          Status CreatedAt  \
0        211131.0  complete        7/1/2016   
1        211133.0  canceled        7/1/2016   
2        211134.0  canceled        7/1/2016   
3        211135.0  complete        7/1/2016   
4        211136.0  order_refunded  7/1/2016   
...           ...             ...       ...   
1048570 NaN        NaN             NaN        
1048571 NaN        NaN             NaN        
1048572 NaN        NaN             NaN        
1048573 NaN        NaN             NaN        
1048574 NaN        NaN             NaN        

                                                                 SKU   Price  \
0        kreations_YI 06-L                                            1950.0   
1        kcc_Buy 2 Frey Air Freshener & Get 1 Kasual Body Spray Free  240.0    
2        Ego_UP0017-999-MR0                                           2450.0   
3        kcc_krone deal                                               360.0    
4        BK7010400AG                                                  555.0    
...              ...                                                    ...    
1048570  NaN                                                         NaN       
1048571  NaN                                                         NaN       
1048572  NaN                                                         NaN       
1048573  NaN                                                         NaN       
1048574  NaN                                                         NaN       

         QuantityOrdered  GrandTotal IncrementID       CategoryName  \
0        1.0              1950.0      100147443   Women's Fashion     
1        1.0              240.0       100147444   Beauty & Grooming   
2        1.0              2450.0      100147445   Women's Fashion     
3        1.0              60.0        100147446   Beauty & Grooming   
4        2.0              1110.0      100147447   Soghaat             
...      ...                 ...            ...       ...             
1048570 NaN              NaN          NaN         NaN                 
1048571 NaN              NaN          NaN         NaN                 
1048572 NaN              NaN          NaN         NaN                 
1048573 NaN              NaN          NaN         NaN                 
1048574 NaN              NaN          NaN         NaN                 

        SalesCommissionCode  ...  Month CustomerSince      MY    FY  \
0        \N                  ...  7.0    2016-7        7-2016  FY17   
1        \N                  ...  7.0    2016-7        7-2016  FY17   
2        \N                  ...  7.0    2016-7        7-2016  FY17   
3        R-FSD-52352         ...  7.0    2016-7        7-2016  FY17   
4        \N                  ...  7.0    2016-7        7-2016  FY17   
...      ..                  ...  ...       ...           ...   ...   
1048570  NaN                 ... NaN     NaN           NaN     NaN    
1048571  NaN                 ... NaN     NaN           NaN     NaN    
1048572  NaN                 ... NaN     NaN           NaN     NaN    
1048573  NaN                 ... NaN     NaN           NaN     NaN    
1048574  NaN                 ... NaN     NaN           NaN     NaN    

        CustomerID  Unnamed: 21  Unnamed: 22 Unnamed: 23 Unnamed: 24  \
0        1.0       NaN          NaN          NaN         NaN           
1        2.0       NaN          NaN          NaN         NaN           
2        3.0       NaN          NaN          NaN         NaN           
3        4.0       NaN          NaN          NaN         NaN           
4        5.0       NaN          NaN          NaN         NaN           
...      ...        ..           ..           ..          ..           
1048570 NaN        NaN          NaN          NaN         NaN           
1048571 NaN        NaN          NaN          NaN         NaN           
1048572 NaN        NaN          NaN          NaN         NaN           
1048573 NaN        NaN          NaN          NaN         Na

Previously, it was mentioned that there is a value \N in the column which we can consider as nan or missing value.

Therefore, we first change the value \N to nan

In [ ]:
df.replace(r'\\N$', np.nan, regex = True, inplace = True)
df.isna().sum()/df.shape[0]*100

ItemID                 44.255394 
Status                 44.257206 
CreatedAt              44.255394 
SKU                    44.257302 
Price                  44.255394 
QuantityOrdered        44.255394 
GrandTotal             44.255394 
IncrementID            44.255394 
CategoryName           45.019670 
SalesCommissionCode    89.667120 
DiscountAmount         44.255394 
PaymentMethod          44.255394 
WorkingDate            44.255394 
BIStatus               44.255394 
 MV                    44.255394 
Year                   44.255394 
Month                  44.255394 
CustomerSince          44.256443 
MY                     44.255394 
FY                     44.255394 
CustomerID             44.256443 
Unnamed: 21            100.000000
Unnamed: 22            100.000000
Unnamed: 23            100.000000
Unnamed: 24            100.000000
Unnamed: 25            100.000000
dtype: float64

Next, we'll see if there are any columns that have no values and can be deleted.

In [ ]:
#persentasi missing value pada setiap kolom
df.isnull().sum()/df.shape[0]*100

ItemID                 44.255394 
Status                 44.257206 
CreatedAt              44.255394 
SKU                    44.257302 
Price                  44.255394 
QuantityOrdered        44.255394 
GrandTotal             44.255394 
IncrementID            44.255394 
CategoryName           45.019670 
SalesCommissionCode    89.667120 
DiscountAmount         44.255394 
PaymentMethod          44.255394 
WorkingDate            44.255394 
BIStatus               44.255394 
 MV                    44.255394 
Year                   44.255394 
Month                  44.255394 
CustomerSince          44.256443 
MY                     44.255394 
FY                     44.255394 
CustomerID             44.256443 
Unnamed: 21            100.000000
Unnamed: 22            100.000000
Unnamed: 23            100.000000
Unnamed: 24            100.000000
Unnamed: 25            100.000000
dtype: float64

From the information above, we can see that there are 5 columns that have 100% missing values. Therefore, these 5 columns will be deleted.

In [ ]:
df.drop(df.columns[[21,22,23,24,25]], axis=1, inplace=True)
df.isna().sum()/df.shape[0]*100

ItemID                 44.255394
Status                 44.257206
CreatedAt              44.255394
SKU                    44.257302
Price                  44.255394
QuantityOrdered        44.255394
GrandTotal             44.255394
IncrementID            44.255394
CategoryName           45.019670
SalesCommissionCode    89.667120
DiscountAmount         44.255394
PaymentMethod          44.255394
WorkingDate            44.255394
BIStatus               44.255394
 MV                    44.255394
Year                   44.255394
Month                  44.255394
CustomerSince          44.256443
MY                     44.255394
FY                     44.255394
CustomerID             44.256443
dtype: float64

Now we can see that all remaining columns have missing values with amounts above 44.25% of the total data. It is possible that the missing values in each of these columns lie in the same row so we will delete them all first.

In [ ]:
df.dropna(how='all', inplace = True)
df.isna().sum()/df.shape[0]*100

ItemID                 0.000000 
Status                 0.003251 
CreatedAt              0.000000 
SKU                    0.003422 
Price                  0.000000 
QuantityOrdered        0.000000 
GrandTotal             0.000000 
IncrementID            0.000000 
CategoryName           1.371030 
SalesCommissionCode    81.463892
DiscountAmount         0.000000 
PaymentMethod          0.000000 
WorkingDate            0.000000 
BIStatus               0.000000 
 MV                    0.000000 
Year                   0.000000 
Month                  0.000000 
CustomerSince          0.001882 
MY                     0.000000 
FY                     0.000000 
CustomerID             0.001882 
dtype: float64

In [ ]:
df.isna().sum()

ItemID                 0     
Status                 19    
CreatedAt              0     
SKU                    20    
Price                  0     
QuantityOrdered        0     
GrandTotal             0     
IncrementID            0     
CategoryName           8014  
SalesCommissionCode    476176
DiscountAmount         0     
PaymentMethod          0     
WorkingDate            0     
BIStatus               0     
 MV                    0     
Year                   0     
Month                  0     
CustomerSince          11    
MY                     0     
FY                     0     
CustomerID             11    
dtype: int64

From the data cleansing process that has been carried out, we can see the remaining missing values as follows:
1. There are 15 missing values in the status column or around 0.002% of the total data
2. There are 20 missing values in the SKU column or around 0.003% of the total data
3. There are 164 missing values in the CategoryName column or around 0.02% of the total data
4. There are 137,175 missing values in the SalesCommissionCode column or around 23.46% of the total data
5. There are 11 missing values in the CustomerSince and CustomerID columns or around 0.001% of the total data

Next, we will carry out a cleansing process for any remaining missing values that still exist

### SalesCommissionCode

We will start the cleansing process from the SalesCommissionCode column because it has 24.46% missing values, which is a very large number and will have a significant impact on the analysis process if deleted directly.

In [ ]:
df['SalesCommissionCode'].unique()

array([nan, 'R-FSD-52352', '105259', ..., 'GRS-4090974-1',
       'GRS-4090975-1', '100559728'], dtype=object)

The SalesCommissionCode column contains the sales commission code. This means that for every sale of goods there is a salesperson who contributes to the sale. Therefore, we can change the missing value in this column to "No Code" assuming the buyer buys the goods directly without going through sales. By using this method, we can utilize the 24% missing values in this column into data that can be analyzed.

In [ ]:
df.fillna({'SalesCommissionCode':'No Code'},inplace=True)
df.isna().sum()

ItemID                 0   
Status                 19  
CreatedAt              0   
SKU                    20  
Price                  0   
QuantityOrdered        0   
GrandTotal             0   
IncrementID            0   
CategoryName           8014
SalesCommissionCode    0   
DiscountAmount         0   
PaymentMethod          0   
WorkingDate            0   
BIStatus               0   
 MV                    0   
Year                   0   
Month                  0   
CustomerSince          11  
MY                     0   
FY                     0   
CustomerID             11  
dtype: int64

### CategoryName

In [ ]:
df['CategoryName'].unique()

array(["Women's Fashion", 'Beauty & Grooming', 'Soghaat',
       'Mobiles & Tablets', 'Appliances', 'Home & Living',
       "Men's Fashion", 'Kids & Baby', nan, 'Others', 'Entertainment',
       'Computing', 'Superstore', 'Health & Sports', 'Books',
       'School & Education'], dtype=object)

In this column, there are 0.02% missing values. this number is relatively small. However, we can still change this missing value rather than delete it.

If we look at the unique value of this column, there is an Others category. This Others category is goods that are not included in other categories.

We will change these missing values to 'Others' because we cannot determine which specific category each missing value falls into.

In [ ]:
df.fillna({'CategoryName':'Others'},inplace=True)
df.isna().sum()

ItemID                 0 
Status                 19
CreatedAt              0 
SKU                    20
Price                  0 
QuantityOrdered        0 
GrandTotal             0 
IncrementID            0 
CategoryName           0 
SalesCommissionCode    0 
DiscountAmount         0 
PaymentMethod          0 
WorkingDate            0 
BIStatus               0 
 MV                    0 
Year                   0 
Month                  0 
CustomerSince          11
MY                     0 
FY                     0 
CustomerID             11
dtype: int64

### Remaining Missing Value

Now, the remaining missing values are only in the status, sku, Customer Since, and Customer ID columns which cannot be changed because the data values are specific and the percentage of missing values is less than 1% so it is better to delete them

In [ ]:
df.dropna(how='any', inplace = True)
df.isna().sum()

ItemID                 0
Status                 0
CreatedAt              0
SKU                    0
Price                  0
QuantityOrdered        0
GrandTotal             0
IncrementID            0
CategoryName           0
SalesCommissionCode    0
DiscountAmount         0
PaymentMethod          0
WorkingDate            0
BIStatus               0
 MV                    0
Year                   0
Month                  0
CustomerSince          0
MY                     0
FY                     0
CustomerID             0
dtype: int64

## Value Grouping

### Status

In [ ]:
df['Status'].unique()

array(['complete', 'canceled', 'order_refunded', 'received', 'refund',
       'closed', 'fraud', 'holded', 'exchange', 'pending_paypal', 'paid',
       'cod', 'pending', 'processing', 'payment_review'], dtype=object)

The status column has many unique values. we will group the status column values. There are 15 types of status which we will group into four groups, namely Complete, Canceled, In Process, and Fraud.

The value included in the **Complete** group is **complete**

The values included in the **Cancel** group are **canceled, order_refunded, refund, and closed**

The values included in the **In Process** group are **received, held, exchange, pending_paypal, paid, cod, pending, processing, and payment_review**

The value that is included in the **Fraud** group is **fraud**

In [ ]:
df['Status'] = df['Status'].replace('complete','Complete')

In [ ]:
df['Status'] = df['Status'].replace('canceled','Cancel').replace('order_refunded','Cancel').replace('refund','Cancel').replace('closed','Cancel')

In [ ]:
df['Status'] = df['Status'].replace('received','In Process').replace('holded','In Process').replace('exchange','In Process').replace('pending_paypal','In Process').replace('paid','In Process').replace('cod','In Process').replace('pending','In Process').replace('processing','In Process').replace('payment_review','In Process')

In [ ]:
df['Status'] = df['Status'].replace('fraud','Fraud')

In [ ]:
df['Status'].value_counts()

Cancel        269293
Complete      233685
In Process    81486 
Fraud         10    
Name: Status, dtype: int64

In [ ]:
df['Status'].value_counts()/df.shape[0]*100

Cancel        46.074419
Complete      39.982104
In Process    13.941766
Fraud         0.001711 
Name: Status, dtype: float64

After grouping the data, we can see that there are Cancel status for 269,293 transactions, Complete for 233,685 transactions, In Process for 81,486 transactions, and Fraud for 10 transactions.

### PaymentMethod

In [ ]:
df['PaymentMethod'].unique()

array(['cod', 'ublcreditcard', 'mygateway', 'customercredit',
       'cashatdoorstep', 'mcblite', 'internetbanking', 'marketingexpense',
       'productcredit', 'financesettlement', 'Payaxis', 'jazzvoucher',
       'jazzwallet', 'Easypay', 'Easypay_MA', 'easypay_voucher',
       'bankalfalah', 'apg'], dtype=object)

From the unique value above, there are 18 payment methods available. The values in payment_method can be grouped into two groups, namely Cash and Digital.

The value that goes into the Cash group is cod, and the rest goes into the Digital group.

we will create a new column containing the Cash and Digital groups

In [ ]:
df['Payment'] = df['PaymentMethod'].apply(lambda x: 'Cash' if x in ['cod'] else 'Digital')

In [ ]:
df['Payment'].value_counts()

Digital    312557
Cash       271917
Name: Payment, dtype: int64

After grouping in the Payment column, we can see digital payment methods 312557 times and cash 271917 times

## Dating Format

Next we will see if any columns have the wrong date format and whether it is necessary to create a new date format column.

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 584474 entries, 0 to 584523
Data columns (total 22 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   ItemID               584474 non-null  float64
 1   Status               584474 non-null  object 
 2   CreatedAt            584474 non-null  object 
 3   SKU                  584474 non-null  object 
 4   Price                584474 non-null  float64
 5   QuantityOrdered      584474 non-null  float64
 6   GrandTotal           584474 non-null  float64
 7   IncrementID          584474 non-null  object 
 8   CategoryName         584474 non-null  object 
 9   SalesCommissionCode  584474 non-null  object 
 10  DiscountAmount       584474 non-null  float64
 11  PaymentMethod        584474 non-null  object 
 12  WorkingDate          584474 non-null  object 
 13  BIStatus             584474 non-null  object 
 14   MV                  584474 non-null  object 
 15  Year             

there are several columns that have date values ranging from CreatedAt, WorkingDate, Year, Month, CustomerSince, MY. of all these columns, none has the datetime data type format. Therefore, let's check the value in each column.

In [ ]:
df.head()

ItemID    Status CreatedAt  \
0  211131.0  Complete  7/1/2016   
1  211133.0  Cancel    7/1/2016   
2  211134.0  Cancel    7/1/2016   
3  211135.0  Complete  7/1/2016   
4  211136.0  Cancel    7/1/2016   

                                                           SKU   Price  \
0  kreations_YI 06-L                                            1950.0   
1  kcc_Buy 2 Frey Air Freshener & Get 1 Kasual Body Spray Free  240.0    
2  Ego_UP0017-999-MR0                                           2450.0   
3  kcc_krone deal                                               360.0    
4  BK7010400AG                                                  555.0    

   QuantityOrdered  GrandTotal IncrementID       CategoryName  \
0  1.0              1950.0      100147443   Women's Fashion     
1  1.0              240.0       100147444   Beauty & Grooming   
2  1.0              2450.0      100147445   Women's Fashion     
3  1.0              60.0        100147446   Beauty & Grooming   
4  2.0              1110.0      100147447   Soghaat             

  SalesCommissionCode  ...  WorkingDate BIStatus      MV     Year Month  \
0  No Code             ...  7/1/2016     #REF!     1,950   2016.0  7.0    
1  No Code             ...  7/1/2016     Gross     240     2016.0  7.0    
2  No Code             ...  7/1/2016     Gross     2,450   2016.0  7.0    
3  R-FSD-52352         ...  7/1/2016     Net       360     2016.0  7.0    
4  No Code             ...  7/1/2016     Valid     1,110   2016.0  7.0    

   CustomerSince      MY    FY CustomerID Payment  
0  2016-7         7-2016  FY17  1.0        Cash    
1  2016-7         7-2016  FY17  2.0        Cash    
2  2016-7         7-2016  FY17  3.0        Cash    
3  2016-7         7-2016  FY17  4.0        Cash    
4  2016-7         7-2016  FY17  5.0        Cash    

[5 rows x 22 columns]

First, we will change the values in the CreatedAt and WorkingDate columns to datetime with the type dd/mm/yyyy

In [ ]:
df['CreatedAt'] = pd.to_datetime(df['CreatedAt'])

In [ ]:
df['WorkingDate'] = pd.to_datetime(df['WorkingDate'])

In [ ]:
df

ItemID      Status  CreatedAt  \
0       211131.0  Complete   2016-07-01   
1       211133.0  Cancel     2016-07-01   
2       211134.0  Cancel     2016-07-01   
3       211135.0  Complete   2016-07-01   
4       211136.0  Cancel     2016-07-01   
...          ...     ...            ...   
584519  905204.0  In Process 2018-08-28   
584520  905205.0  In Process 2018-08-28   
584521  905206.0  In Process 2018-08-28   
584522  905207.0  In Process 2018-08-28   
584523  905208.0  In Process 2018-08-28   

                                                                SKU     Price  \
0       kreations_YI 06-L                                            1950.0     
1       kcc_Buy 2 Frey Air Freshener & Get 1 Kasual Body Spray Free  240.0      
2       Ego_UP0017-999-MR0                                           2450.0     
3       kcc_krone deal                                               360.0      
4       BK7010400AG                                                  555.0      
...             ...                                                    ...      
584519  WOFSCE5AE00357AECDE                                          699.0      
584520  MATHUA5AF70A7D1E50A                                          35599.0    
584521  MATSAM5B6D7208C6D30                                          129999.0   
584522  MATSAM5B1509B4696EA                                          87300.0    
584523  MATSAM5B10F91A9B6AB                                          108640.0   

        QuantityOrdered  GrandTotal IncrementID       CategoryName  \
0       1.0              1950.0      100147443   Women's Fashion     
1       1.0              240.0       100147444   Beauty & Grooming   
2       1.0              2450.0      100147445   Women's Fashion     
3       1.0              60.0        100147446   Beauty & Grooming   
4       2.0              1110.0      100147447   Soghaat             
...     ...                 ...            ...       ...             
584519  1.0              849.0       100562385   Women's Fashion     
584520  1.0              35899.0     100562386   Mobiles & Tablets   
584521  2.0              652178.0    100562387   Mobiles & Tablets   
584522  2.0              652178.0    100562387   Mobiles & Tablets   
584523  2.0              652178.0    100562387   Mobiles & Tablets   

       SalesCommissionCode  ...  WorkingDate BIStatus        MV     Year  \
0       No Code             ... 2016-07-01    #REF!     1,950     2016.0   
1       No Code             ... 2016-07-01    Gross     240       2016.0   
2       No Code             ... 2016-07-01    Gross     2,450     2016.0   
3       R-FSD-52352         ... 2016-07-01    Net       360       2016.0   
4       No Code             ... 2016-07-01    Valid     1,110     2016.0   
...         ...             ...        ...      ...        ...       ...   
584519  No Code             ... 2018-08-28    Valid     699       2018.0   
584520  No Code             ... 2018-08-28    Gross     35,599    2018.0   
584521  No Code             ... 2018-08-28    Gross     259,998   2018.0   
584522  No Code             ... 2018-08-28    Gross     174,600   2018.0   
584523  No Code             ... 2018-08-28    Gross     217,280   2018.0   

       Month  CustomerSince      MY    FY CustomerID  Payment  
0       7.0   2016-7         7-2016  FY17  1.0        Cash     
1       7.0   2016-7         7-2016  FY17  2.0        Cash     
2       7.0   2016-7         7-2016  FY17  3.0        Cash     
3       7.0   2016-7         7-2016  FY17  4.0        Cash     
4       7.0   2016-7         7-2016  FY17  5.0        Cash     
...     ...      ...            ...   ...  ...         ...     
584519  8.0   2018-8         8-2018  FY19  115320.0   Cash     
584520  8.0   2018-8         8-2018  FY19  115326.0   Digital  
584521  8.0   2018-7         8-2018  FY19  113474.0   Digital  
584522  8.0   2018-7         8-2018  FY19  113474.0   Digital  
584523  8.0   2018-7         8-2018  FY19  113474.0   Digital 

Next, we see that the Year, Month columns contain the Float data type. We will change it to an integer data type

In [ ]:
df['Year'] = df['Year'].astype(int)

In [ ]:
df['Month'] = df['Month'].astype(int)

Next, we will change the MY and CustomerSince columns to a datetime data type with mm/yyyy format

In [ ]:
df['MY'] = pd.to_datetime(df['MY'])
df['MY'] = df['MY'].dt.strftime('%m/%Y')

In [ ]:
df['CustomerSince'] = pd.to_datetime(df['CustomerSince'])
df['CustomerSince'] = df['CustomerSince'].dt.strftime('%m/%Y')

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 584474 entries, 0 to 584523
Data columns (total 22 columns):
 #   Column               Non-Null Count   Dtype         
---  ------               --------------   -----         
 0   ItemID               584474 non-null  float64       
 1   Status               584474 non-null  object        
 2   CreatedAt            584474 non-null  datetime64[ns]
 3   SKU                  584474 non-null  object        
 4   Price                584474 non-null  float64       
 5   QuantityOrdered      584474 non-null  float64       
 6   GrandTotal           584474 non-null  float64       
 7   IncrementID          584474 non-null  object        
 8   CategoryName         584474 non-null  object        
 9   SalesCommissionCode  584474 non-null  object        
 10  DiscountAmount       584474 non-null  float64       
 11  PaymentMethod        584474 non-null  object        
 12  WorkingDate          584474 non-null  datetime64[ns]
 13  BIStatus      

In [ ]:
df.head()

ItemID    Status  CreatedAt  \
0  211131.0  Complete 2016-07-01   
1  211133.0  Cancel   2016-07-01   
2  211134.0  Cancel   2016-07-01   
3  211135.0  Complete 2016-07-01   
4  211136.0  Cancel   2016-07-01   

                                                           SKU   Price  \
0  kreations_YI 06-L                                            1950.0   
1  kcc_Buy 2 Frey Air Freshener & Get 1 Kasual Body Spray Free  240.0    
2  Ego_UP0017-999-MR0                                           2450.0   
3  kcc_krone deal                                               360.0    
4  BK7010400AG                                                  555.0    

   QuantityOrdered  GrandTotal IncrementID       CategoryName  \
0  1.0              1950.0      100147443   Women's Fashion     
1  1.0              240.0       100147444   Beauty & Grooming   
2  1.0              2450.0      100147445   Women's Fashion     
3  1.0              60.0        100147446   Beauty & Grooming   
4  2.0              1110.0      100147447   Soghaat             

  SalesCommissionCode  ...  WorkingDate BIStatus      MV   Year Month  \
0  No Code             ... 2016-07-01    #REF!     1,950   2016  7      
1  No Code             ... 2016-07-01    Gross     240     2016  7      
2  No Code             ... 2016-07-01    Gross     2,450   2016  7      
3  R-FSD-52352         ... 2016-07-01    Net       360     2016  7      
4  No Code             ... 2016-07-01    Valid     1,110   2016  7      

   CustomerSince       MY    FY CustomerID Payment  
0  07/2016        07/2016  FY17  1.0        Cash    
1  07/2016        07/2016  FY17  2.0        Cash    
2  07/2016        07/2016  FY17  3.0        Cash    
3  07/2016        07/2016  FY17  4.0        Cash    
4  07/2016        07/2016  FY17  5.0        Cash    

[5 rows x 22 columns]

# Data Analysis

After carrying out the data preparation process, we will carry out data analysis. First, we will find out the existing information from the data we have

### Payment Method

In [ ]:
#In Percentage
Payment_pct = df['Payment'].value_counts()/df.shape[0]*100
Payment_pct

Digital    53.47663
Cash       46.52337
Name: Payment, dtype: float64

### Transaction Status

In [ ]:
#In Percentage
Status_pct = df['Status'].value_counts()/df.shape[0]*100
Status_pct

Cancel        46.074419
Complete      39.982104
In Process    13.941766
Fraud         0.001711 
Name: Status, dtype: float64

### Product Category

In [ ]:
#In Percentage
Category_pct = df['CategoryName'].value_counts()/df.shape[0]*100
Category_pct

Mobiles & Tablets     19.797117
Men's Fashion         15.777947
Women's Fashion       10.217734
Appliances            8.967550 
Superstore            7.461581 
Beauty & Grooming     7.099204 
Others                6.364868 
Soghaat               5.819078 
Home & Living         4.534676 
Entertainment         4.503537 
Health & Sports       2.994487 
Kids & Baby           2.821169 
Computing             2.726041 
School & Education    0.595065 
Books                 0.319946 
Name: CategoryName, dtype: float64

From the information above, we can know that:

1. Digital payment methods were the largest payment method at 53.47% of total transactions while cash payment methods were 46.52% of total transactions.
2. The transaction status cancel is the largest, namely 46.07% of the total transactions. followed by complete transaction status at 39.98%, in process at 13.94% and fraud at 0.001% of total transactions.
3. The 5 most frequent product categories are Mobiles & Tablets at 19.79%, followed by Men's Fashion at 15.77%, Women's Fashion at 10.21%, Appliances at 8.96%, and Superstore at 7.46%.

### Analysis of Payment Methods with Transaction Status

In [ ]:
df_StatusPayment = pd.crosstab(df['Status'],df['Payment'])
df_StatusPayment['Total']=df_payment.sum(axis=1)
df_StatusPayment.sort_values('Total',ascending=False)

Payment       Cash  Digital   Total
Status                             
Cancel      76428   192865   269293
Complete    148039  85646    233685
In Process  47450   34036    81486 
Fraud       0       10       10

From the information above, we can know that:

1. Transactions using the cash payment method are the transactions with the highest complete status, 148,039 transactions
2. Transactions using digital payment methods are transactions with the most cancellation status, 192,865 transactions
3. Transactions with fraud status are all transactions using digital payment methods.

### Analysis of Payment Methods by Item Category

In [ ]:
df_CategoryPayment = pd.crosstab(df['CategoryName'],df['Payment'])
df_CategoryPayment.sort_values('CategoryName',ascending=False)

Payment              Cash  Digital
CategoryName                      
Women's Fashion     37866  21854  
Superstore          13446  30165  
Soghaat             22060  11951  
School & Education  2482   996    
Others              6929   30272  
Mobiles & Tablets   35818  79891  
Men's Fashion       61515  30703  
Kids & Baby         10460  6029   
Home & Living       15138  11366  
Health & Sports     12313  5189   
Entertainment       5038   21284  
Computing           5412   10521  
Books               1491   379    
Beauty & Grooming   26011  15482  
Appliances          15938  36475

From the information above, we can know that:

1. The 5 categories of goods that use the cash payment method the most are Men's Fashion with 61,515 transactions, followed by Women's Fashion with 37,866 transactions, Mobiles & Tablets with 35,818 transactions, Beauty & Grooming with 26,011 transactions, and Soghaat with 33,060 transactions.

2. The 5 categories of goods that use the most digital payment methods are Mobiles & Tablets with 79,891 transactions, followed by Appliances with 36,475 transactions, Men's Fashion with 30,703 transactions, Others with 30,272 transactions, and Superstore with 30,165 transactions.

### Analysis of Transaction Status by Item Category

In [ ]:
df_CategoryStatus = pd.crosstab(df['CategoryName'],df['Status'])
df_CategoryStatus.sort_values('CategoryName',ascending=False)

Status              Cancel  Complete  Fraud  In Process
CategoryName                                           
Women's Fashion     25349   23684     0      10687     
Superstore          16400   16693     0      10518     
Soghaat             9865    19792     0      4354      
School & Education  1301    1584      0      593       
Others              24510   6341      0      6350      
Mobiles & Tablets   65433   41658     6      8612      
Men's Fashion       38996   40291     0      12931     
Kids & Baby         6276    7025      0      3188      
Home & Living       10272   11153     1      5078      
Health & Sports     5854    8049      0      3599      
Entertainment       14819   9327      2      2174      
Computing           8686    5735      1      1511      
Books               530     910       0      430       
Beauty & Grooming   14053   20734     0      6706      
Appliances          26949   20709     0      4755

From the information above, we can know that:

1. The 5 categories of goods with the most canceled transaction statuses are Mobiles & Tablets with 65,433 transactions, followed by Men's Fashion, Appliances, Women's Fashion, and Others.

2. The 5 categories of goods with the highest total transaction status are Mobiles & Tablets with 41,658 transactions, followed by Men's Fashion, Women's Fashion, Beauty & Grooming, and Appliances.

3. The category of goods with the highest fraud transaction status is Mobiles & Tablets with 6 transactions, followed by Entertainment with 2 transactions, and Home & Living and Computing with 1 transaction each.

4. The 5 categories of goods with the highest transaction status in process are Men's Fashion with 12931 transactions, followed by Women's Fashion, Superstore, Mobiles & Tablets, and Beauty & Grooming.

### Income Analysis

We will first define income and losses.


* Income is the total of grand_total which has the status "complete".
* Lost revenue is the total of grand_total which has the status "cancel".
* Potential additional income is the total of grand_total which has the status "in process".
* Total revenue is Revenue + Potential Revenue (assuming all transactions will be "Complete") - Lost Revenue

In [ ]:
# Income
pendapatan = df.loc[(df["Status"] == "Complete"), "GrandTotal"].sum()

print("Pendapatan yang didapat sebanyak:", pendapatan)

Pendapatan yang didapat sebanyak: 1171776013.6705


In [ ]:
# Lost Revenua
pendapatan_lost = df.loc[(df["Status"] == "Cancel"), "GrandTotal"].sum()

print("Pendapatan yang hilang sebanyak:", pendapatan_lost)

Pendapatan yang hilang sebanyak: 3352240872.9059997


In [ ]:
# Potential Additional Value
pendapatan_potensi = df.loc[(df["Status"] == "In Process"), "GrandTotal"].sum()

print("Potensi pendapatan yang akan didapat sebanyak:", pendapatan_potensi)

Potensi pendapatan yang akan didapat sebanyak: 461526689.98999995


In [ ]:
# Total Revenue
total_pendapatan = pendapatan + pendapatan_potensi - pendapatan_lost
print("Total pendapatan sebanyak:", total_pendapatan)

Total pendapatan sebanyak: -1718938169.2454996


### Income Analysis with Cash Payment Method

In [ ]:
# Pendapatan
pendapatan_cash = df.loc[(df["Payment"] == "Cash") & (df["Status"] == "Complete"), "GrandTotal"].sum()

print("Total pendapatan dengan metode pembayaran cash sebanyak:", pendapatan_cash)

Total pendapatan dengan metode pembayaran cash sebanyak: 393423699.62999994


In [ ]:
# Pendapatan yang Hilang
pendapatan_lost_cash = df.loc[(df["Payment"] == "Cash") & (df["Status"] == "Cancel"), "GrandTotal"].sum()

print("Total pendapatan yang hilang dengan metode pembayaran cash sebanyak:", pendapatan_lost_cash)

Total pendapatan yang hilang dengan metode pembayaran cash sebanyak: 508782487.415


In [ ]:
# Potensi Pendapatan
pendapatan_potensi_cash = df.loc[(df["Payment"] == "Cash") & (df["Status"] == "In Process"), "GrandTotal"].sum()

print("Total potensi pendapatan dengan metode pembayaran cash sebanyak:", pendapatan_potensi_cash)

Total potensi pendapatan dengan metode pembayaran cash sebanyak: 163083023.70499998


In [ ]:
# Total Pendapatan
total_pendapatan_cash = pendapatan_cash + pendapatan_potensi_cash - pendapatan_lost_cash
print("Total pendapatan dengan metode pembayaran cash sebanyak:", total_pendapatan_cash)

Total pendapatan dengan metode pembayaran cash sebanyak: 47724235.9199999


From the information above, we can know that:

1. Income from transactions using the cash payment method was 393,423,699.62

2. Lost income from transactions using the cash payment method was 508,782,487.41

3. Potential income from transactions using the cash payment method is 163,083,023.70

4. Total income from transactions using the cash payment method was 47,724,235.91

### Income Analysis with Digital Payment Methods

In [ ]:
# Pendapatan
pendapatan_digital = df.loc[(df["Payment"] == "Digital") & (df["Status"] == "Complete"), "GrandTotal"].sum()

print("Total pendapatan dengan metode pembayaran digital sebanyak:", pendapatan_digital)

Total pendapatan dengan metode pembayaran digital sebanyak: 778352314.0405


In [ ]:
# Pendapatan yang Hilang
pendapatan_lost_digital = df.loc[(df["Payment"] == "Digital") & (df["Status"] == "Cancel"), "GrandTotal"].sum()

print("Total pendapatan yang hilang dengan metode pembayaran digital sebanyak:", pendapatan_lost_digital)

Total pendapatan yang hilang dengan metode pembayaran digital sebanyak: 2843458385.4909997


In [ ]:
# Potensi Pendapatan
pendapatan_potensi_digital = df.loc[(df["Payment"] == "Digital") & (df["Status"] == "In Process"), "GrandTotal"].sum()

print("Total potensi pendapatan dengan metode pembayaran digital sebanyak:", pendapatan_potensi_digital)

Total potensi pendapatan dengan metode pembayaran digital sebanyak: 298443666.285


In [ ]:
# Total Pendapatan
total_pendapatan_digital = pendapatan_digital + pendapatan_potensi_digital - pendapatan_lost_digital
print("Total pendapatan dengan metode pembayaran digital sebanyak:", total_pendapatan_digital)

Total pendapatan dengan metode pembayaran digital sebanyak: -1766662405.1654997


From the information above, we can know that:

1. Income from transactions using digital payment methods was 778,352,314..04

2. Lost income from transactions using digital payment methods was 2,843,458,385.49

3. Potential income from transactions using digital payment methods is 298,443,666.28

4. Total income from transactions using digital payment methods was -1,766,662,405.16

# Conclusion & Recommendation

## Conclusion

1. Digital payment methods are the payment methods most frequently chosen by consumers, accounting for 53.47% of total transactions. However, this method is also the method with the most cancel status of all transactions. The large number of cancellations in this transaction caused the company to lose revenue of 1,766,662,405.16.

2. Meanwhile, the cash payment method is the second option for consumers for 46.52% of total transactions. Even though there are fewer than digital payment methods, this method has the highest complete status of all transactions. With so many complete statuses, the company earned revenue of 47,724,235.91.

3. Digital payment methods are most widely used by consumers when they want to buy goods in the Mobiles & Tablets category. Meanwhile, the cash payment method is most widely used by consumers when they want to buy goods in the Men's Fashion category.

## Recommendation

DIGITAL PAYMENTS

1. There are several possibilities that cause many transactions to be canceled using digital payment methods, such as consumers taking too long to complete payments, failures in bank or financial service provider payment systems, or a digital payment ecosystem that does not provide many choices. (source: razorpay.com)

2. To overcome this problem, companies can provide several solutions such as providing more digital payment options such as credit cards, e-wallets and local bank transfers. Apart from that, the company can also provide suggestions for the same payment method from previous successful transactions.

CASH PAYMENT

1. Cancel status on cash transactions can be caused by several things, such as the consumer's absence when the courier arrives at the location to make the payment or the consumer does not provide money according to the nominal payment.

2. To overcome this problem, companies can propose an agreement with consumers regarding the payment date that will be made when the consumer checks out the goods. Apart from that, companies can also provide cash payment limits to consumers by looking at the average successful transactions using the cash method. (source: pakistantoday.com.pk)

By providing solutions to existing problems, companies can optimize business activities with consumers. Moreover, with the Pakistani government currently encouraging business digitalization, companies can also support the government by continuing to optimize their services. (source: "e-Commerce Policy of Pakistan", Government of Pakistan Commerce Division, Oct 2019)